File: rest-common/src/main/java/org/chronopolis/rest/entities/PersistableEntity.java
package org.chronopolis.rest.entities;

import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.MappedSuperclass;

/**
 * Mark an Entity as Persistable and return the id
 *
 * @author shake
 */
@MappedSuperclass
public class PersistableEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    protected Long id;

    public Long getId() {
        return id;
    }

    public PersistableEntity setId(Long id) {
        this.id = id;
        return this;
    }

}

File: rest-entities/src/main/kotlin/org/chronopolis/rest/kot/entities/PersistableEntity.kt
package org.chronopolis.rest.kot.entities

import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.MappedSuperclass

@MappedSuperclass
open class PersistableEntity(
        @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long = 0
)
